src/Intervention/Image/Image.php

```diff
@@ -129,13 +129,18 @@ public function encode($format = null, $quality = 90)
     public function save($path = null, $quality = null)
     {
         $path = is_null($path) ? $this->basePath() : $path;
+        $dir = substr($path, 0, strrpos($path, '/'));
 
         if (is_null($path)) {
             throw new Exception\NotWritableException(
                 "Can't write to undefined path."
             );
         }
```

Comment:
As I also said, you have to put your added line under the `if (is_null($path)) {...}` part.

That's because if `$path` is null, it would run the string functions on `null`...